man/fi_mr.3.md

```diff
@@ -776,6 +777,27 @@ field is determined by the value specified through iface.
 ## hmem_data
 The hmem_data field is reserved for future use and must be null.
 
+## page_size
+Page size allows applications to notify the provider the page size to be
+used for an MR allocation. Typically, providers can select the optimal
+page size. In cases where a VA range has zero pages backing it, the
+provider may not know the optimal page size during registration. Rather
+than always use a potential inferior page size, allow apps to specify
+the page size to be used.
+
+Page size must be power of 2. If the provider supports the page size
+argument and later determines that this page size cannot be used, RDMA
+operations may fail. For example, suppose the user sets page size to 1
+GiB but the corresponding pages are only 2 MiB. The MR may be configured
+for 1 GiB and the provider cannot change MR page size without
+reconfiguring the MR.
+
+Providers will treat page size as a hint. Meaning that if a provider
+does not support user defined page size, it will be ignored.
```

Comment:
This seems wrong.  If the provider cannot support the size, the call should fail.  The provider support may be to fall back to a less efficient size, but it must be supported.

Reply:
I see where you are coming from. The problem this is trying to solve is how do apps know if a provider supports this field? Or do apps have to be updated with provider specific information knowing which providers may support this?

Reply:
The change is updating the API.  The providers need to support the call being made with and without this field present, and it may be present, but uninitialized by the app.  The providers can handle this based on the API version selected by the app when opening the domain.

Apps should expect all providers to support this field.  The behavior describe by this change is that this value allows the provider to optimize for a specific page size.  Without that knowledge, the provider works the same as it does today and may be less efficient.

I don't think any check is needed by the app to see if the provider implements support.  If a provider doesn't it simply ignores that the field exists.